docs/products/clickhouse/reference/table-functions.rst

```diff
@@ -0,0 +1,31 @@
+Table functions supported in Aiven for ClickHouse®
+==================================================
+
+Table functions return data of the table type and are used to construct tables. Aiven for ClickHouse® supports the following `table functions <https://clickhouse.com/docs/en/sql-reference/table-functions>`_:
+
+* ``cluster``
+* ``clusterAllReplicas``
+* ``cosn``
+* ``deltaLake``
```

Comment:
@joelynch I omitted dictionary on purpose given what you told me on slack! We can modify the doc later down the line when we work more on the feature :)